src/Compiler/FSComp.txt

```diff
@@ -1672,4 +1672,5 @@ featureEscapeBracesInFormattableString,"Escapes curly braces before calling Form
 3555,chkInstanceLetBindingOnStaticClasses,"If a type uses both [<Sealed>] and [<AbstractClass>] attributes, it means it is static. Instance let bindings are not allowed."
 3556,chkImplementingInterfacesOnStaticClasses,"If a type uses both [<Sealed>] and [<AbstractClass>] attributes, it means it is static. Implementing interfaces is not allowed."
 3557,chkAbstractMembersDeclarationsOnStaticClasses,"If a type uses both [<Sealed>] and [<AbstractClass>] attributes, it means it is static. Abstract member declarations are not allowed."
-3558,chkExplicitFieldsDeclarationsOnStaticClasses,"If a type uses both [<Sealed>] and [<AbstractClass>] attributes, it means it is static. Explicit field declarations are not allowed."
\ No newline at end of file
+3558,chkExplicitFieldsDeclarationsOnStaticClasses,"If a type uses both [<Sealed>] and [<AbstractClass>] attributes, it means it is static. Explicit field declarations are not allowed."
+3559,typrelNeverRefinedAwayFromTop,"A type inference variable has been implicitly inferred to have type `obj`. Consider adding explicit type annotations. This warning is off by default and has been explicitly enabled for this project. You may suppress this warning by using #nowarn \"3559\"."
```

Comment:
I believe the last but one sentence can become untrue with time and the last sentence is generally applicable to whatever warning - so I'd remove those for brevity.

Reply:
This wording was Don's (https://github.com/dotnet/fsharp/pull/13298#discussion_r899036766) - I do agree with you. Is the new version better?

Reply:
Hmhm not sure. Honestly I don't think it's good to suggest disabling the warning in any scenario but I might not have enough context here. @abelbraaksma?

Reply:
Disabling a warning as a suggestion in a warning is common practice, quite a few warnings already have that hint. I wouldn’t be too worried about making that part of the message.

Though, about wording, “a type inference variable” is not really a thing. Though I understand its meaning, there may not be a variable at all (like, try `Ok “”`, the Error type ends up `obj`, but won’t have a variable).

Perhaps: “a type inference variable or free type parameter”? (Or without the word “free”)

Reply:
Honestly just "a type" seems unambiguous to me.

Reply:
Good point, the word “inferred” is already in the sentence. 

> A type has been implicitly inferred ….